tpm2tools/signer_test.go

```diff
@@ -0,0 +1,87 @@
+package tpm2tools
+
+import (
+	"crypto"
+	"crypto/rsa"
+	"crypto/sha1"
+	"crypto/sha256"
+	"testing"
+
+	"github.com/google/go-tpm-tools/internal"
+	"github.com/google/go-tpm/tpm2"
+)
+
+func rsaTemplate(hash tpm2.Algorithm) tpm2.Public {
+	template := AIKTemplateRSA(nil)
+	// Can't sign arbitrary data if restricted.
+	template.Attributes &= ^tpm2.FlagRestricted
+	template.RSAParameters.Sign.Hash = hash
+	return template
+}
+
+func TestSign(t *testing.T) {
+	rwc := internal.GetTPM(t)
+	defer CheckedClose(t, rwc)
+
+	tests := []struct {
+		name     string
+		hash     crypto.Hash
```

Comment:
Were you thinking the crypto.Hash member should be public? I was thinking it would be private, but I don't really like using private members in tests.

Reply:
We'd talked about this offline a bit, and we'd decided on the TpmSigner becoming private but having its fields public. I'm going to leave the hash field in the test struct though, as the TpmSigner hash field still isn't part of the public API, so shouldn't be used in unit tests.